src/api/odometry/odomMath.cpp

```diff
@@ -14,24 +14,47 @@ OdomMath::OdomMath() = default;
 
 OdomMath::~OdomMath() = default;
 
-QLength OdomMath::computeDistanceToPoint(const Point2d &ipoint, const OdomState &istate) {
-  const double xDiff = (ipoint.x - istate.x).convert(meter);
-  const double yDiff = (ipoint.y - istate.y).convert(meter);
-  return std::sqrt((xDiff * xDiff) + (yDiff * yDiff)) * meter;
+QLength OdomMath::computeDistanceToPoint(const Point2d &ipoint,
+                                         const OdomState &istate,
+                                         const StateMode &ipointMode) {
+  if (ipointMode == StateMode::CARTESIAN) {
```

Comment:
This is a little bit redundant. How about:
```cpp
const double xDiff = (ipoint.inMode(ipointMode).y - istate.x).convert(meter);
const double yDiff = (ipoint.inMode(ipointMode).x - istate.y).convert(meter);
return std::sqrt((xDiff * xDiff) + (yDiff * yDiff)) * meter;
```
The same goes for `computeAngleToPoint` and `computeDistanceAndAngleToPoint`

Reply:
I am still working on it.